python/python/bystro/utils/tar.py

```diff
@@ -0,0 +1,76 @@
+"""Utility functions for safely invoking tar in cross-OS manner."""
+
+import subprocess
+from enum import Enum
+from subprocess import CalledProcessError
+from typing import NoReturn
+
+
+class OperatingSystem(Enum):
+    """Represent the operating system on which the current Python process is running."""
+
+    linux = "linux"
+    macosx = "macosx"
+
+
+# here and throughout, fully qualify executable names to avoid privilege escalation
+
+
+def _get_executable_name(name: str) -> str:
+    try:
+        result = subprocess.run(
```

Comment:
Could we instead use shuttle.which for portability?

Reply:
good call, thanks